File: Zipping and Unzipping files in .NET application/[VB]-Zipping and Unzipping files in .NET application/VB/SynchronizeFolders.vb
Imports ComponentPro.IO

Namespace ArchiveManager
	Partial Public Class SynchronizeFolders
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal recursive As RecursionMode, ByVal syncDateTime As Boolean, ByVal syncAttributes As Boolean, ByVal comparisonMethod As Integer, ByVal checkForResumability As Boolean, ByVal searchPattern As String, ByVal sourceDir As String)
			InitializeComponent()

			chkRecursive.Checked = recursive = RecursionMode.RecurseIntoAllSubFolders
			chkUpdateTime.Checked = syncDateTime
			chkUpdateAttributes.Checked = syncAttributes
			cbxComparisonType.SelectedIndex = comparisonMethod
			chkResumability.Checked = checkForResumability
			txtSearchPattern.Text = searchPattern
			txtSourceDir.Text = sourceDir
		End Sub

		Public ReadOnly Property Recursive() As RecursionMode
			Get
				If chkRecursive.Checked Then
					Return RecursionMode.RecurseIntoAllSubFolders
				Else
					Return RecursionMode.None
				End If
			End Get
		End Property

		Public ReadOnly Property SyncDateTime() As Boolean
			Get
				Return chkUpdateTime.Checked
			End Get
		End Property

		Public ReadOnly Property SyncAttributes() As Boolean
			Get
				Return chkUpdateAttributes.Checked
			End Get
		End Property

		Public ReadOnly Property ComparisonMethod() As Integer
			Get
				Return cbxComparisonType.SelectedIndex
			End Get
		End Property

		Public ReadOnly Property SearchPattern() As String
			Get
				Return txtSearchPattern.Text
			End Get
		End Property

		Public ReadOnly Property SourceDir() As String
			Get
				Return txtSourceDir.Text
			End Get
		End Property

		Public ReadOnly Property CheckForResumability() As Boolean
			Get
				Return chkResumability.Checked
			End Get
		End Property

		Public ReadOnly Property DeleteFiles() As Boolean
			Get
				Return chkDeleteFiles.Checked
			End Get
		End Property

		Private Sub cbxComparisonType_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles cbxComparisonType.SelectedIndexChanged
			chkResumability.Visible = cbxComparisonType.SelectedIndex = 1
		End Sub

		Private Sub btnSourceDir_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnSourceDir.Click
			Try
				Dim dlg As New FolderBrowserDialog()
				dlg.Description = "Select local folder"
				dlg.SelectedPath = txtSourceDir.Text
				If dlg.ShowDialog() = System.Windows.Forms.DialogResult.OK Then
					txtSourceDir.Text = dlg.SelectedPath
				End If
			Catch exc As System.Exception
				Util.ShowError(exc)
			End Try
		End Sub
	End Class
End Namespace

File: Zipping and Unzipping files in .NET application/[C#]-Zipping and Unzipping files in .NET application/C#/SynchronizeFolders.cs
using System.Windows.Forms;
using ComponentPro.IO;

namespace ArchiveManager
{
    public partial class SynchronizeFolders : Form
    {
        public SynchronizeFolders()
        {
            InitializeComponent();
        }

        public SynchronizeFolders(RecursionMode recursive, bool syncDateTime, bool syncAttributes, int comparisonMethod, bool checkForResumability, string searchPattern, string sourceDir)
        {
            InitializeComponent();

            chkRecursive.Checked = recursive == RecursionMode.RecurseIntoAllSubFolders;
            chkUpdateTime.Checked = syncDateTime;
            chkUpdateAttributes.Checked = syncAttributes;
            cbxComparisonType.SelectedIndex = comparisonMethod;
            chkResumability.Checked = checkForResumability;
            txtSearchPattern.Text = searchPattern;
            txtSourceDir.Text = sourceDir;
        }

        public RecursionMode Recursive
        {
            get { return chkRecursive.Checked ? RecursionMode.RecurseIntoAllSubFolders : RecursionMode.None; }
        }

        public bool SyncDateTime
        {
            get { return chkUpdateTime.Checked; }
        }

        public bool SyncAttributes
        {
            get { return chkUpdateAttributes.Checked; }
        }

        public int ComparisonMethod
        {
            get { return cbxComparisonType.SelectedIndex; }
        }

        public string SearchPattern
        {
            get { return txtSearchPattern.Text; }
        }

        public string SourceDir
        {
            get { return txtSourceDir.Text; }
        }

        public bool CheckForResumability
        {
            get { return chkResumability.Checked; }
        }

        public bool DeleteFiles
        {
            get { return chkDeleteFiles.Checked; }
        }

        private void cbxComparisonType_SelectedIndexChanged(object sender, System.EventArgs e)
        {
            chkResumability.Visible = cbxComparisonType.SelectedIndex == 1;
        }

        private void btnSourceDir_Click(object sender, System.EventArgs e)
        {
            try
            {
                FolderBrowserDialog dlg = new FolderBrowserDialog();
                dlg.Description = "Select local folder";
                dlg.SelectedPath = txtSourceDir.Text;
                if (dlg.ShowDialog() == System.Windows.Forms.DialogResult.OK)
                {
                    txtSourceDir.Text = dlg.SelectedPath;
                }
            }
            catch (System.Exception exc)
            {
                Util.ShowError(exc);
            }
        }
    }
}